internal/pkg/config/config.go

```
@@ -377,3 +411,16 @@ func getValueOrEnv(secretValue *string, envKey, defaultValue string) string {
 	return getEnv(envKey, defaultValue)
 }
 
+// buildBaseURL constructs the base URL if not provided
+func (c *Config) buildBaseURL() {
+	if c.BaseURL == "" {
+		if c.Environment == "local" || c.Environment == "development" || c.Environment == "dev" {
+			c.BaseURL = fmt.Sprintf("http://localhost:%s", c.Port)
+		} else {
+			// In production, BaseURL should be explicitly configured
+			// This is a fallback that should be overridden with the actual domain
+			c.BaseURL = fmt.Sprintf("https://your-domain.com")
```

golangci-lint (1.64.8)

S1039: unnecessary use of fmt.Sprintf

(gosimple)